### ISSUE TITLE
Add a 'lazy' training mode to YatNMN: freeze kernel, train only alpha + epsilon

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#37]
## Summary

Add a **"lazy" training mode to `YatNMN`** where the **kernel (and bias) are frozen** and only **`alpha` and `epsilon`** are trainable.

The idea: keep the learned/initialised feature directions (`kernel`) fixed and let the network adapt **only** the per-neuron scale (`alpha`) and the distance-softening (`epsilon`). This is the "lazy" regime — fixed features, only the scaling/normalisation parameters move — and is useful for cheap adaptation, probing the role of α/ε, and grokking-dynamics studies.

## Current behavior

In `nmn/nnx/layers/nmn.py`, `YatNMN` declares everything as `nnx.Param`, so all of it trains:

```python
self.kernel        = nnx.Param(...)               # feature directions
self.bias          = nnx.Param(...) | None
self.alpha         = nnx.Param(...) | None         # learnable scale (use_alpha + constant_alpha=None)
self.epsilon_param = nnx.Param(...) | None         # learnable only if learnable_epsilon=True
```

There is **no flag to freeze the kernel**. `constant_alpha`/`use_alpha` control α, and `learnable_epsilon` controls ε, but the kernel is always a trainable `nnx.Param`. To get "freeze kernel, train α+ε" today, a user has to reach outside the layer and build a custom optimizer/grad filter over the NNX param tree — there's no first-class, discoverable way.

## Proposed API

A single opt-in flag, e.g. `lazy: bool = False` (or `freeze_kernel: bool = False`):

```python
layer = YatNMN(
    in_features=512, out_features=512,
    use_alpha=True,            # learnable alpha
    learnable_epsilon=True,    # learnable epsilon
    lazy=True,                 # <- freeze kernel (+bias); only alpha & epsilon train
    rngs=nnx.Rngs(0),
)
```

When `lazy=True`:
- `kernel` (and `bias`) become **non-trainable** so a standard `nnx.state(model, nnx.Param)` optimizer skips them;
- `alpha` and `epsilon_param` remain trainable `nnx.Param`s (so it composes with `use_alpha`/`learnable_epsilon`).

### Implementation options
1. **Frozen variable type** — store the frozen tensors under a dedicated non-`Param` variable type (e.g. an `nnx.Variable` subclass `FrozenParam`) so `nnx.state(model, nnx.Param)` naturally excludes them. Cleanest for optimizers that filter on `nnx.Param`.
2. **`stop_gradient` in `__call__`** — wrap `kernel`/`bias` reads in `jax.lax.stop_gradient` when `lazy=True`. Simple, but the params still appear in the `Param` tree (they'd just get zero grads), so it doesn't reduce optimizer state unless combined with a filter.

Option 1 is preferable since it also removes the frozen weights from optimizer moment buffers.

## Why in the library

Freezing the kernel while training α/ε is a YatNMN-specific regime (it's meaningful precisely because of the `y²/(‖q−k‖²+ε)·α` structure). Doing it correctly per-backend (nnx Param filtering vs torch `requires_grad_(False)` vs mlx) is exactly the kind of thing the layer should own, rather than every downstream model re-deriving it.

Happy to send a PR (option 1) for the `nnx` backend, with matching `freeze`/`requires_grad` handling for torch/mlx.

### Environment
- `nmn` 0.3.0, `flax` 0.12.x (NNX), JAX 0.9.x

**Comment by mlnomadpy** [comment]
Follow-up parity defect found in the completed implementation: NNX uniquely treats `freeze_kernel=False` as an override that disables `lazy=True`:

```python
m = YatNMN(8, 4, lazy=True, freeze_kernel=False, rngs=nnx.Rngs(0))
assert m.lazy is False  # current NNX behavior
```

PyTorch, Linen, Keras, TensorFlow, and MLX use logical OR semantics, matching their docs that either true flag freezes the kernel. The NNX test `test_freeze_kernel_false_overrides` currently codifies the divergence.

Suggested follow-up: use `lazy or freeze_kernel` in NNX and replace that test with cross-backend alias-precedence parity.
